### ISSUE TITLE
Why are the results always true for this example in the engineering project

### ISSUE THREAD

**Issue by css521** [manifold-systems/manifold-sample-project#4]
Why are the results always true for this example in the engineering project? Wasn't the likes method redefined as `(Function<String, Boolean>)(thing -> !thing.equals("relaxation"))`?

```
private static void useMapStructuralInterface() {
    out.println("\n\n### Use Maps with Structural Interfaces ###\n");
    // Use a Map to directly back any interface
    Dog dog = (Dog)new HashMap<>();
    dog.setBreed("Foxhound");
    ((Map)dog).put("likes", (Function<String,Boolean>)(thing -> !thing.equals("relaxation")));
    String breed = dog.getBreed();
    out.println(breed);
    out.println(dog.likes("ball"));
    out.println(dog.likes("relaxation"));
  }
  @Structural
  interface Dog {
    String getBreed();
    void setBreed(String breed);
    default boolean likes(String thing) {
      return true;
    }
  }
```
Run result ：
![image](https://user-images.githubusercontent.com/42376488/230287424-55de2048-3519-402f-9afc-319e01b0c92a.png)

**Comment by rsmckinney** [comment]
This is a bug dealing with default methods on interfaces. Dog#likes is a default method that returns true. It is called instead of the lambda. A fix for this issue will be in the next release this weekend. Thanks for reporting this!

**Comment by rsmckinney** [comment]
Fix available with release 2023.1.5.  See https://github.com/manifold-systems/manifold/issues/436.
